ci/mkpipeline.py

```diff
@@ -110,6 +110,14 @@ def main() -> int:
     with open(Path(__file__).parent / args.pipeline / "pipeline.template.yml") as f:
         raw = f.read()
     raw = raw.replace("$RUST_VERSION", rust_version())
+
+    # On 'main' or tagged branches, we use a separate remote cache that only CI can write to.
+    if os.environ["BUILDKITE_BRANCH"] == "main" or os.environ["BUILDKITE_TAG"]:
+        bazel_remote_cache = "https://bazel-remote-pa.dev.materialize.com"
+    else:
+        bazel_remote_cache = "https://bazel-remote.dev.materialize.com"
```

Comment:
So anyone can upload to this cache? This sounds like an easy way to compromise all developer systems.

Reply:
What do you mean by anyone? Any engineer at Materialize or anyone in the world?

Reply:
Anyone in the world. Is there some kind of SSO or credentials required to upload to this cache?

Reply:
Ah, just saw this:

> Adds a TeleportProxy class in Python to spawn a proxy using tsh, this allows developers to connect to the remote cache running in our infra.

Should have checked the rest of the PR, my bad!

Reply:
For a developer to access the `https://bazel-remote.dev.materialize.com` cache you need Teleport access, CI can also access it. `https://bazel-remote-pa.dev.materialize.com` is only accessible via CI